pkgs/applications/graphics/unigine_heaven/default.nix

```diff
@@ -0,0 +1,66 @@
+{ lib,
+  stdenv,
+  fetchurl,
+  makeWrapper,
+  libX11,
+  libXext,
+  libXrandr,
+  freetype,
+  fontconfig,
+  libXrender,
+  libXinerama,
+  autoPatchelfHook,
+  libGL,
+  openal,
+}:
+
+let
+  version = "4.0";
+in
+
+stdenv.mkDerivation {
+  name = "Unigine_Heaven";
+  pname = "Unigine_Heaven";
+
+  src = fetchurl {
+    url = "https://assets.unigine.com/d/Unigine_Heaven-${version}.run";
+    sha256 = "19rndwwxnb9k2nw9h004hyrmr419471s0fp25yzvvc6rkd521c0v";
+  };
+
+  unpackPhase = "true";
+
+  opl = openal; #need for wrapProgram
+  lgl = libGL;
+
+  installPhase = ''
+    cp $src extract.run
+    chmod +x ./extract.run
+    ./extract.run --target $out
```

Comment:
No need to make an extra copy of the sources just to add the executable flag. The .run file is actually a shell script (with an embedded tar archive) so you can simply do this instead:
```
sh $src --target $out
```

This way you don't have to set +x on the file, since the sh interpreter just reads it normally instead of asking the system to execute it. (the system would look at the file and see that it is a script for sh and just ask sh to read and interpret it anyway)